components/io-libs/adios2/SPECS/adios2.spec

```diff
@@ -0,0 +1,159 @@
+# This RPM .spec file is part of the OpenHPC project.
+#
+# It may have been modified from the default version supplied by the underlying
+# release package (if available) in order to apply patches, perform customized
+# build/install configurations, and supply additional files to support
+# desired integration conventions.
+#
+#----------------------------------------------------------------------------eh-
+
+# Build that is dependent on compiler/mpi toolchains
+%define ohpc_compiler_dependent 1
+%define ohpc_mpi_dependent 1
+%define ohpc_python_dependent 1
+%global python_family python3
+%include %{_sourcedir}/OHPC_macros
+
+# Base package name
+%define pname adios2
+
+Summary: The Adaptable IO System v2 (ADIOS2)
+Name:    %{pname}-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
+Version: 2.8.3
+Release: 1%{?dist}
+License: Apache License 2.0
+Group:   %{PROJ_NAME}/io-libs
+Url:     https://adios2.readthedocs.io/en/latest/index.html
+Source0: https://github.com/ornladios/ADIOS2/archive/refs/tags/v%{version}.tar.gz
+AutoReq: no
+
+%if 0%{?rhel} || 0%{?openEuler}
+BuildRequires:  bzip2-devel
+%endif
+%if 0%{?suse_version}
+BuildRequires:  libbz2-devel
+%endif
+
+BuildRequires: libtool cmake make
+Requires:      lmod%{PROJ_DELIM} >= 7.6.1
+BuildRequires: phdf5-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
+Requires:      phdf5-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
+
+BuildRequires: %{python_prefix}-devel %{python_prefix}-setuptools
+BuildRequires: %{python_prefix}-numpy-%{compiler_family}%{PROJ_DELIM}
+BuildRequires: %{python_prefix}-mpi4py-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
+
+# Default library install path
+%define install_path %{OHPC_LIBS}/%{compiler_family}/%{mpi_family}/%{pname}/%version
+
+%description
+The Adaptable IO System (ADIOS2) provides a simple, flexible way for
+scientists to describe the data in their code that may need to be
+written, read, or processed outside of the running simulation. By
+providing an external to the code XML file describing the various
+elements, their types, and how you wish to process them this run, the
+routines in the host code (either Fortran or C) can transparently change
+how they process the data.
+
+%prep
+%setup -q -n %{PNAME}-%{version}
+
+%build
+mkdir adios2-build
+cd adios2-build
+
+# OpenHPC compiler/mpi designation
+%ohpc_setup_compiler
+
+module load phdf5
+module load %{python_module_prefix}numpy
+module load %{python_module_prefix}mpi4py
+
+export CC=mpicc
+export CXX=mpicxx
+export F77=mpif77
+export FC=mpif90
+export MPICC=mpicc
+export MPIFC=mpif90
+export MPICXX=mpicxx
+
+cmake \
+    -DCMAKE_INSTALL_PREFIX=%{install_path} \
+    -DBUILD_SHARED_LIBS=ON \
+    -DADIOS2_USE_MPI=ON \
+    -DADIOS2_USE_HDF5=ON -DHDF5_ROOT="${HDF5_DIR}" \
+    -DADIOS2_USE_ZeroMQ=OFF \
+    -DADIOS2_USE_Fortran=ON \
+    -DADIOS2_USE_Python=ON \
+    -DADIOS2_USE_SST=ON \
+    -DADIOS2_USE_BZip2=ON \
+    -DBUILD_TESTING=ON \
+    ..
+make
+# make test
```

Comment:
Should we run the tests here ?
There are 2 issues with them:
1) some tests fail
2) the tests execution takes about an hour

Reply:
I decided not to run ADIOS2's tests but to copy few of them and use them as smoke tests in OpenHPC tests suite.